src/include/common/WavefrontSender.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <list>
+#include <set>
+#include <boost/uuid/uuid.hpp>
+#include "common/HistogramGranularity.h"
+
+namespace wavefront {
+    /**
+    * An abstract WavefrontSender class that contains various sending methods along with flushing and closing logic
+    *
+    * @author Mengran Wang (mengranw@vmware.com)
+    */
+    class WavefrontSender {
+    public:
+        WavefrontSender() {}
+
+        ~WavefrontSender() {}
+
+        /**
+         * Sends the given metric to Wavefront
+         *
+         * @param name      The name of the metric. Spaces are replaced with '-' (dashes) and quotes will
+         *                  be automatically escaped.
+         * @param value     The value to be sent.
+         * @param timestamp The timestamp in milliseconds since the epoch to be sent. If equal -1 then the
+         *                  timestamp is assigned by Wavefront when data is received.
+         * @param source    The source (or host) that's sending the metric. If empty then assigned by
+         *                  Wavefront.
+         * @param tags      The tags associated with this metric.
+         */
+        virtual void
+        sendMetric(const std::string &name, double value, long timestamp = -1, const std::string &source = "",
```

Comment:
you should also support sendDelta method
see https://github.com/wavefrontHQ/wavefront-java-sdk/blob/master/src/main/java/com/wavefront/sdk/entities/metrics/WavefrontMetricSender.java#L51